Automattic/jetpack
# Stats: Add API for stats page module status management

Fixes #31056

## Proposed changes:

The PR proposes to add two API to update and get module status for Stats pages, so that user can hide certain modules(cards) on certain pages. The APIs are created as  `wpcom/v2` endpoints, and are intended to be used by Jetpack site locally and by Calypso as well. After the changes are deployed to WPCOM, we'll then wire up the APIs for simple sites and also configure the requests to be forwarded for Jetpack sites on WPCOM.

Please note only modules of the corresponding page would be handled: https://github.com/Automattic/jetpack/pull/31057/files#diff-64e2998c3517113f1facb45f92a6a4ddcbbeb5d8af31e0e0fe42570761129df2R21

### POST `wpcom/v2/stats-admin/modules`

Sample request is as below, where 'traffic' is the page name, and 'videos' is the module name. When a false value passed as the value of a module, the module would be turned off.
```JSON
{
  "traffic": {"videos": false, "app-promo": true, ... },
  ...
}
```

Sample response:
```JSON
{"updated": true}
```

### GET `wpcom/v2/stats-admin/modules`

Sample response:
```JSON
{
  "traffic": {"videos": false, "app-promo": true, ... },
  "insights": {"year-in-review": false},
  ...
}
```



More information about `wpcom/v2` endpoints: PCYsg-aqU-p2.

### Other information:


## Jetpack product discussion
p1HpG7-lZo-p2


## Does this pull request change what data or activity we track or use?
no

## Testing instructions:

* Choose a http client you favour. I use [REST Client for VSCode](https://marketplace.visualstudio.com/items?itemName=humao.rest-client).
* Update modules sample request
```
POST https://your-site.au.ngrok.io/wp-json/wpcom/v2/stats-admin/modules
Content-Type: application/json; charset=UTF-8
Cookie: copy-cookie-from-api-requests-from-dev-network-tools
X-WP-Nonce: copy-nonce-from-api-requests-from-dev-network-tools

{"traffic":{"videos":true}}
```
* Get modules status sample request:
```
GET https://your-site.au.ngrok.io/wp-json/wpcom/v2/stats-admin/modules
Content-Type: application/json; charset=UTF-8
Cookie: copy-cookie-from-api-requests-from-dev-network-tools
X-WP-Nonce: copy-nonce-from-api-requests-from-dev-network-tools
```
- Ensure the module is enabled or disabled accordingly
- Ensure only the modules defined are allowed
- Ensure Unit Tests pass
